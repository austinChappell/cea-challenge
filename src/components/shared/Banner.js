import React from 'react';

import CountDown from './CountDown';

const Banner = (props) => {
  const {
    countDownMessage,
    nextEventTime,
    subTitle,
    title,
  } = props;

  return (
    <div className="Banner">
      <h1>
        {title}
      </h1>
      <h4>{subTitle}</h4>
      <CountDown
        eventTime={nextEventTime}
        message={countDownMessage}
      />
    </div>
  );
};

export default Banner;
