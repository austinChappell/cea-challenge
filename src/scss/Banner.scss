.Banner {
  background-color: $color-primary;
  text-align: center;
  padding: 20px;
  .go-back {
    position: absolute;
    top: 10px;
    left: 10px;
  }
  * {
    color: $color-white;
  }
}