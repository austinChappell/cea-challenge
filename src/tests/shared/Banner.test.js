import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Banner from '../../components/shared/Banner';

const props = {
  countDownMessage: '',
  goBackButton: true,
  goBackText: '',
  goBackRoute: '',
  nextEventTime: new Date(1522030848733),
  subTitle: '',
  title: '',
};

describe('The Banner Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Banner {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
