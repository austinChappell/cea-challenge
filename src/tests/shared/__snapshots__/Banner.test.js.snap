// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Banner Component should render correctly 1`] = `
<div
  className="Banner"
>
  <span
    className="go-back"
  >
    <Link
      href=""
      replace={false}
      to=""
    />
  </span>
  <h1 />
  <h4 />
  <CountDown
    eventTime={2018-03-26T02:20:48.733Z}
    message=""
    startTime={null}
  />
</div>
`;
